src/App.js

```diff
@@ -2,77 +2,50 @@ import React, { Component } from 'react';
 import './App.css';
 
 class App extends Component {
-  _isMounted = false;
-
   state = {
-    source: 'https://i.imgur.com/VYUdUNJ.png',
+    source: '',
     error: null,
+    loading: true,
   };
 
-  fetchCat() {
-    fetch("https://aws.random.cat/meow").then(res=>res.json()).then(
-      (result) => {
-        this.setState({
-          source: result.file
-        });
-    },
-    (error) => {
-      this.setState({
-        error
-      });
-    }
-    )
-  }
-
   componentDidMount() {
-    this._isMounted = true;
-    const { source } = this.state;
-    this.fetchCat(source);
-
+    this.fetchCat();
   }
 
-  componentWillUnmount() {
-    this._isMounted = false;
+  fetchCat = async () => {
+    try {
+      const result = await fetch("https://aws.random.cat/meow");
+      const image = result.json();
+      this.setState({ source: image.file, loading: false });
+    } catch (e) {
+      this.setState({ error: e.message, loading: false });
+    }
```

Comment:
This is the async/await version of almost exactly what you were doing above... The only difference is the loading state is added, so that whenever the data is returned, even if there is an error, it is set to false